### ISSUE TITLE
Wrong permissions for the downloaded toolchain

### ISSUE THREAD

**Issue by adrienluitot** [pine64/ArduinoCore-bouffalo#1]
The binaries of the toolchain for Linux (`riscv64-unknown-elf-linux-0.0.1.zip`) have bad permissions and require some modifications from the user. They don't have the executable bit.

These binaries are located in: 
- `riscv64-unknown-elf/0.0.1/bin/`
- `riscv64-unknown-elf/0.0.1/riscv64-unknown-elf/bin/`
- `riscv64-unknown-elf/0.0.1/libexec/gcc/riscv64-unknown-elf/8.3.0/`

Thanks to @9names for pointing me these paths. 

As the purpose is, I think, to simplify things, it might be better to change the permissions in the downloaded archive.

I don't know if MacOS is also impacted, and I suppose that Windows isn't.

**Comment by gamelaster** [comment]
Will take look, thanks

**Comment by pfeerick** [comment]
I'll be looking into this this week, in https://github.com/pfeerick/ArduinoCore-bouffalo-packages/issues/1

I think the underlying issue is the choice of archive... a zip was used instead of a tar, which would preserve permissions.

**Comment by adrienluitot** [comment]
This could be good indeed, though if a zip file is needed, I think that if the archive is made on linux, it keeps the permissions.

**Comment by pfeerick** [comment]
Yes, whilst the zip format itself doesn't support file permissions, it
seems with the right flags at time of zip, and of unzip, infozip will
preserve them in another field... the question would then be does the IDE
support that... only one sure fire way to find out!!!

**Comment by gamelaster** [comment]
@pfeerick the reason why is everything hosted on PINE64 Cluster is because many times Github CDN is terribly slow and I wait ages until it downloads something. Anyway, I think we can have multiple "mirrors". I'm open to merge your work @pfeerick here.

**Comment by adrienluitot** [comment]
I've made some tests few days ago, and without special flags the permissions seemed to be kept after a `zip` and an `unzip`, but I don't know if there are default parameters on my distro. :shrug: 

For the hosting, maybe the Pine64 cluster could regularly pull the files from the Github repo ?
So we would have the speed of Pine64, and the last updates of Github.

**Comment by pfeerick** [comment]
I'm not complaining about the files being on the pine64 cluster, more that they are a black box at the moment, with no idea as to the source of the files, or how the packages were put together ;) For example, where do the toolchains come from? Because knowting that, the next question is, is it possible to get an arm64 toolchain happening as well? i.e. lets get a little pinebook pro love happening as well ;) 

Hence why I'd ideally like to automate generation of the binaries via a github action if practical, and which also means the upload / generation of the files is all done on the github side, and then simply mirrored to the pine64 cluster if that's faster / more reliable at times. Plus it makes the builds reproducible.  

The other aspect to this is once I know enough about the full system, I can look at getting a PlatformIO platform package going also...

**Comment by danielbarry** [comment]
Just to add that this problem still exists in Linux, it is needed to run:

```
cd ~/.arduino15/packages/bouffalo/tools/riscv64-unknown-elf/0.0.1/bin
chmod +x *
cd ~/.arduino15/packages/bouffalo/tools/riscv64-unknown-elf/0.0.1/libexec/gcc/riscv64-unknown-elf/8.3.0
chmod +x *
```

**Comment by technoblogy** [comment]
I had to run the equivalent commands on Mac, but compilation then failed with:

    clang: error: unsupported option '--traditional-format'
    exit status 1

**Comment by gamelaster** [comment]
@pfeerick well, you made really good points, having a mirror is good. I will think about this.

**Comment by MartijnBraam** [comment]
Somehow still an issue
